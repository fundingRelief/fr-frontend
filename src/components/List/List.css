.List {
  font-family: 'Comfortaa', cursive;
  a {
    text-decoration: none;
  }

  a:link {
    color: rgba(40, 167, 118, 0.6);
  }

  a:visited {
    color: rgba(40, 167, 118, 0.6);
  }

  a:hover {
    color: rgb(50, 150, 207);
  }

  a:active {
    color: rgba(40, 167, 118, 0.6);
  }
}

.header {
  font-family: 'Comfortaa', cursive !important;
  padding-top: 1em;
}

.location {
  color: teal !important;
  font-family: 'Comfortaa', cursive !important;
  font-weight: bold;
  padding-top: .5em;
  padding-bottom: .5em;
}

.description {
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;  
  overflow: hidden;
  font-family: 'Comfortaa', cursive;
}

.donation {
  padding-top: 1.5em;
  padding-bottom: .5em !important;
}

.progressBar{
  margin-bottom: 0 !important;
}

.raised {
  color:black !important;
  padding-top: .5em !important;
}



