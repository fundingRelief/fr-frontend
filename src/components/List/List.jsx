import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './List.css';
import { Card, Icon, Image, Progress } from 'semantic-ui-react';

const List = ({
  id,
  campaign_name,
  current_amount,
  goal,
  percentage_raised,
  location,
  img_url,
  link_url,
  description,
  last_donation,
}) => {

  const numberWithCommas = (x) => {
    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  };

  return (
    <Card 
      color="teal"
      raised={true}>
      <Card.Content>
        <Link className={styles.link} to={`/campaigns/${id}`}>
          <Image 
            bordered rounded size="large"
            style={{ backgroundImage: `url(${img_url})` }}
            className={styles.cardImg}
          />
        </Link>
        <Link className={styles.link} to={`/campaigns/${id}`}>
          <Card.Header 
            className={styles.header}
            textAlign={'left'}
          >        
            {campaign_name}
          </Card.Header>
        </Link>
 
        <Card.Description className={styles.location}>
          {location}
        </Card.Description>
        <Card.Description className={styles.description}>
          {description}
        </Card.Description>
        <Card.Meta className={styles.donation}>
          {last_donation}
        </Card.Meta>
        <Card.Meta>
          <Progress 
            percent={percentage_raised} 
            progress
            className={styles.progressBar}
            color="teal"/>
        </Card.Meta>
        <Card.Meta className={styles.raised}>
          <strong>${numberWithCommas(current_amount)} raised of </strong> ${numberWithCommas(goal)}{' '}goal
        </Card.Meta>
      </Card.Content>
      <Card.Content 
        extra
        textAlign="center">
        <a href={link_url} target="_blank" rel="noopener noreferrer">
          <Icon name="user" />
              Donate Now at GoFundMe
        </a>
      </Card.Content>
    </Card>
  );
};

List.propTypes = {
  id: PropTypes.number.isRequired,
  campaign_name: PropTypes.string.isRequired,
  current_amount: PropTypes.number.isRequired,
  goal: PropTypes.number.isRequired,
  percentage_raised: PropTypes.number.isRequired,
  location: PropTypes.string.isRequired,
  img_url: PropTypes.string.isRequired,
  link_url: PropTypes.string.isRequired,
  description: PropTypes.string,
  last_donation: PropTypes.string.isRequired,
};

export default List;
