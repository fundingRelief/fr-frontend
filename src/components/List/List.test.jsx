import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(
      <List
        id={1}
        campaign_name="Our parents lost their home to the Almeda fire"
        current_amount={944}
        goal={5000}
        percentage={18.8}
        donors={18}
        location="PORTLAND, OR"
        img_url="https://d2g8igdw686xgo.cloudfront.net/51544330_1601348243982129_r.jpeg"
        link_url="https://gf.me/u/y2z2h8"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
