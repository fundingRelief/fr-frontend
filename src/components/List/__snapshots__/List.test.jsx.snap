// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<Card
  color="teal"
  raised={true}
>
  <CardContent>
    <Image
      as="img"
      bordered={true}
      className="cardImg"
      href="https://gf.me/u/y2z2h8"
      rel="noopener noreferrer"
      rounded={true}
      size="large"
      style={
        Object {
          "backgroundImage": "url(https://d2g8igdw686xgo.cloudfront.net/51544330_1601348243982129_r.jpeg)",
        }
      }
      target="_blank"
      ui={true}
    />
    <CardHeader
      className="header"
      href="https://gf.me/u/y2z2h8"
      rel="noopener noreferrer"
      target="_blank"
      textAlign="left"
    >
      Our parents lost their home to the Almeda fire
    </CardHeader>
    <CardDescription
      className="location"
    >
      PORTLAND, OR
    </CardDescription>
    <CardDescription
      className="description"
    >
      On September 8th our parents and many other families lost their home in Phoenix, Oregon. Our parents lost their place to relax and escape to after work. We as a family lost the place we knew we could walk right in without knocking and raid the fridge. The place where my boys would go to and run around in the backyard and play with Paco our family dog. The place where my brother would come back home to every summer after teaching in Oklahoma as teacher for Teach for America. The place where my sister would come home to when she would get homesick or just need a break from being a student at Portland State University. We decided to start this GoFundMe for our parents who have sacrificed for us their entire lives, they always put our needs ahead of theirs even to this day even after losing their home. We hope that we can do our part for them now. Thank you.
    </CardDescription>
    <CardMeta
      className="donation"
    >
      Last donation 2h ago
    </CardMeta>
    <CardMeta>
      <Progress
        className="progressBar"
        color="teal"
        percent={18}
        progress={true}
      />
    </CardMeta>
    <CardMeta
      className="raised"
    >
      <strong>
        $
        944
         raised of 
      </strong>
       $
      5,000
       
      goal
    </CardMeta>
  </CardContent>
  <CardContent
    extra={true}
    textAlign="center"
  >
    <a
      href="https://gf.me/u/y2z2h8"
      rel="noopener noreferrer"
      target="_blank"
    >
      <Icon
        as="i"
        name="user"
      />
      Donate Now at GoFundMe
    </a>
  </CardContent>
</Card>
`;
